### DISCUSSION TITLE
Official documentation of how to create an msbuild task delivered with nuget

### DISCUSSION THREAD

**Discussion by msedi** [NuGet/Home#12078] [Q&A]
Hello,

We have an msbuild task that we deliver via nuget. I had to search a lot through the web to find out specific details but couldn't find an official documentation of how do it. Is there anything?

Thanks
Martin

**Comment by zivkan** [comment]
https://docs.microsoft.com/en-us/nuget/concepts/msbuild-props-and-targets

**Reply by msedi** [reply]
@zivkan: Thanks for the link, but I was more looking like of what is possible and what is necessary (pitfalls etc) when distributing an msbuild task. Something like https://natemcmaster.com/blog/2017/07/05/msbuild-task-in-nuget/ which is quite nice but a bit outdated.
